cmd/collector/collector.go

```diff
@@ -173,13 +181,38 @@ func run() error {
 		}
 	}()
 
+	var server *http.Server
+	// Start the HTTP server in a separate goroutine
+	go func() {
+		mux := http.NewServeMux()
+		mux.HandleFunc("/records", flowRecordHandler)
+		mux.HandleFunc("/reset", resetRecordHandler)
+
+		server = &http.Server{
+			Addr:              ":8080",
+			Handler:           mux,
+			ReadHeaderTimeout: 5 * time.Second,
+		}
+
+		klog.InfoS("Server listening on :8080")
+		if err := server.ListenAndServe(); err != http.ErrServerClosed {
+			klog.ErrorS(err, "ListenAndServe error")
+		}
+	}()
+
 	stopCh := make(chan struct{})
 	go signalHandler(stopCh, messageReceived)
 
 	<-stopCh
 	// Stop the collector process
 	cp.Stop()
 	klog.Info("Stopping IPFIX collector")
+	// Attempt to gracefully shut down the server
+	if err := server.Shutdown(context.Background()); err != nil {
```

Comment:
here you probably want a context with a 5s timeout, just for the shutdown operation

Reply:
Added. Thanks.